flutter-engine/src/loader.rs

```diff
@@ -0,0 +1,67 @@
+use crate::FlutterAotSnapshot;
+use libloading::Library;
+use std::io::Error;
+use std::path::Path;
+use xmas_elf::sections::SectionData;
+use xmas_elf::symbol_table::Entry;
+use xmas_elf::ElfFile;
+
+pub unsafe fn load_snapshot(aot_path: &Path) -> Result<(Library, FlutterAotSnapshot), Error> {
+    let mut vm_snapshot_data_size = 0;
```

Comment:
It might be worth adding a check to ensure the sizes aren't zero.

Reply:
are we sure that they can never be zero?